rlclientlib/logger/async_batcher.h

```diff
@@ -153,9 +164,37 @@ namespace reinforcement_learning { namespace logger {
 
     RETURN_IF_FAIL(collection_serializer.finalize(status));
 
+    if (_events_counter_status == events_counter_status::ENABLE) {
+      _buffer_end_event_number = evt.get_event_number();
+      *original_count = (_buffer_end_event_number - _buffer_start_event_number);
+      if (_queue.size() == 0) {
+        //reset counter values if queue is empty to prevent overflow incase of too many events
+        *original_count+= reset_counter() - _buffer_end_event_number;
+        _buffer_start_event_number = 0;
+        _buffer_end_event_number = 0;
+      }
+    }
+
     return error_code::success;
   }
 
+  template<typename TEvent, template<typename> class TSerializer>
+  void async_batcher<TEvent, TSerializer>::increment_counter()  {
+    std::unique_lock<std::mutex> mlock(_counter_mutex);
+    if (_events_counter_status == events_counter_status::ENABLE) { ++_event_number; }
+  }
+
+  template<typename TEvent, template<typename> class TSerializer>
+  unsigned int async_batcher<TEvent, TSerializer>::reset_counter() {
```

Comment:
Maybe change the name to reset_counter_get_count?

Reply:
updated